tests/python-pytest/onnx/test_operators.py

```diff
@@ -1183,6 +1183,15 @@ def test_onnx_export_take_raise(tmp_path, dtype, axis):
     op_export_test('take', M, [x, y], tmp_path)
 
 
+# onnxruntime currently only supports float32 and int64
+@pytest.mark.parametrize("dtype", ["float32", "int64"])
+@pytest.mark.parametrize("depth", [1, 3, 5, 10])
+def test_onnx_export_one_hot(tmp_path, dtype, depth):
+    M = def_model('one_hot', depth=depth, dtype=dtype)
+    x = mx.random.randint(0, 10, (depth * depth)).astype('int64')
```

Comment:
shall we use a separate parameter to control the shape?

Reply:
Also I think mxnet one_hot can also take non-int64 input, in that case we might need to add a `Cast` op before 'OneHot'

Reply:
Thanks for the feedback. It looks like it is already cast to int64 based on the onnx documentation (https://github.com/onnx/onnx/blob/master/docs/Changelog.md#OneHot-11):

`In case 'indices' is of non-integer type, the values will be casted to int64 before use.`

Reply:
i see thanks！

Reply:
Added a shape parameter.